src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/RunCodeActionService.cs

```diff
@@ -71,6 +86,29 @@ public class RunCodeActionService : BaseCodeActionService<RunCodeActionRequest,
             };
         }
 
+        private static string GetNewFilePath(string newFileName, string currentFilePath)
+        {
+            var directory = Path.GetDirectoryName(currentFilePath);
+            return Path.Combine(directory, newFileName);
+        }
+
+        bool IsRenameDocumentOperation(CodeActionOperation o, out DocumentId oldDocumentId, out DocumentId newDocumentId, out string name)
+        {
+            var type = Type.GetType("Microsoft.CodeAnalysis.CodeActions.RenameDocumentOperation, Microsoft.CodeAnalysis.Workspaces, Version=2.4.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35");
```

Comment:
I don't think you need to include the version number, culture or publickeytoken. This assembly should already be loaded and I'd hate to see this fail (and be another place to update) each time we update Roslyn.

Reply:
Actually, you might be able to just grab it from `Configuration.RoslynWorkspaces` ([src](https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Abstractions/Configuration.cs#L12)).

See, https://github.com/OmniSharp/omnisharp-roslyn/blob/e18913e887144119c41d60f1842e49f8e9bfcf72/src/OmniSharp.Roslyn/MetadataHelper.cs for how we typically handle reflection scenarios in OmniSharp. We also have some reflection extension methods which are intended to fail quickly so we can easily tell when Roslyn internal surface area breaks something ((https://github.com/OmniSharp/omnisharp-roslyn/blob/e18913e887144119c41d60f1842e49f8e9bfcf72/src/OmniSharp.Abstractions/Utilities/ReflectionExtensions.cs) ).

Reply:
yeah +1 for using the existing lazy extensions